app/javascript/components/common/TrackIcon.tsx

```diff
@@ -1,6 +1,11 @@
 import React from 'react'
 
-export function TrackIcon({ track }) {
+type TrackIconProps = {
+  iconUrl: string
+  title: string
+}
+
+export function TrackIcon({ track }: { track: TrackIconProps }) {
```

Comment:
I would have expected less nesting (because of the naming convention ComponentProps):

```typescript
export function TrackIcon({ iconUrl, title }: TrackIconProps) {
```